File: divsum/divsum.py
# -*- coding: utf-8 -*-
import math

for t in range(int(input())):
    n = int(input())
    print(sum([ i + (n // i if n // i < n and n // i != i else 0) for i in range(1, int(math.sqrt(n)) + 1) if n != 1 and n % i == 0 ]))

File: divsum/divsum.cc
#include <iostream>
#include <cmath>

using std::cin;
using std::cout;
using std::cerr;
using std::endl;

int main() {
    int t, n;
    cin >> t;

    while (t--) {
        cin >> n;
        int sum = 0;

        for (int i = 1; i <= sqrt(n) && i < n; ++i) {
            sum += (n % i == 0 ? i + ((n / i) != i && (n / i) < n ? n / i : 0) : 0);
        }

        cout << sum << endl;
    }

    return 0;
}
